tracer.go

```diff
@@ -437,7 +437,7 @@ func (t *Tracer) emitNewSpanMetrics(sp *Span, newTrace bool) {
 }
 
 func (t *Tracer) reportSpan(sp *Span) {
-	if !sp.isSamplingFinalized() {
+	if !sp.context.isSamplingFinalized() {
```

Comment:
is reportSpan called under sp.Lock ?

Reply:
nope.  Added `FinishWithOptions` to the test and locks in `reportSpan()`.  At this point I'm wondering if it's better just to add a 
```
s.Lock()
defer s.Unlock()
```
at the top of `FinishWithOptions`.  Running benchmarks.

Reply:
Nevermind.  `FinishWithOptions` has callbacks.  That's a bad idea.